src/Build.UnitTests/Evaluation/Expander_Tests.cs

```diff
@@ -3394,12 +3396,6 @@ public void PropertyFunctionStaticMethodIntrinsicMaths()
             result = expander.ExpandIntoStringLeaveEscaped(@"$([MSBuild]::Modulo(2345.5, 43))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance);
 
             Assert.Equal((2345.5 % 43).ToString(), result);
-
-            // test for overflow wrapping
-            result = expander.ExpandIntoStringLeaveEscaped(@"$([MSBuild]::Add(9223372036854775807, 20))", ExpanderOptions.ExpandProperties, MockElementLocation.Instance);
-
-            double expectedResult = 9223372036854775807D + 20D;
-            Assert.Equal(expectedResult.ToString(), result);
```

Comment:
FYI @ghogen - if this get's approved and merged in the current proposed functionality, we should probably explicitly document this here https://learn.microsoft.com/en-us/visualstudio/msbuild/property-functions#msbuild-property-functions, with advice to indicate double literal if scale over precision is preferred (as this is now prioritizing precision over scale, by respecting integer literals as longs)